### PR TITLE
os/memstore: misc fix in memstore

### PR DESCRIPTION
**Pull request by mslovy**

### CODE REVIEW

File: src/os/memstore/MemStore.cc [hunk 1]
```diff
@@ -1384,6 +1384,13 @@ int MemStore::_destroy_collection(const coll_t& cid)
     cp->second->exists = false;
   }
   used_bytes -= cp->second->used_bytes();
+
+  // remove data file if it is existed
+  string fn = path + "/" + stringify(cp->first);
+  int r = ::unlink(fn.c_str());
+  if (r < 0 && r != -ENOENT)
+    return r;
+
```
Comment (liewegas) [inline src/os/memstore/MemStore.cc]: should this be guarded by the new option?

Comment (mslovy) [inline src/os/memstore/MemStore.cc]: I don't think this should be guarded by new option.  The case is:

1) create pool with pgs,  memstore_persistent_collections = true now
2) then stop the osd, and the collection file will be saved
3) start osd then with memstore_persistent_collections = false
4) delete the pool and destroy pg

if guarded by option, the stale *_head will stay in the data_dir

**Comment by mslovy** [comment]
is that meanful?  I think someone may use it as a temporary high speed memory cache in the future and may not need any data when the cluster restart.  If so, I will rebase it to master, Otherwise I will close it.

**Comment by cbodley** [comment]
this might be tricky to use in practice, because the `ceph-osd --mkfs` step initializes the `coll_t::meta()` collection, which needs to be present when the `ceph-osd` daemon actually starts up. you might need a special case to persist only that collection

**Comment by liewegas** [comment]
Yeah.. that's pretty much exactly why memstore does the persistence thing on clean shutdown.  @mslovy if you don't have a concrete, specific need for this let's close it until you (or someone) does?

**Comment by mslovy** [comment]
one of the cases I want to use it as a distributed in-memory cache such as redis in order to cache data and also easier to replicated between nodes and scale-out the whole clusters.  it will provides a persistent radosclient API and would be easy to use as a kv-cache or object cache.
However, I find that it is hard to repair the pg state when the whole cluster is restarted unexpectedly. So this may solve a part of problem to allow all collections to lose it state and rollback to a new clear pg.

**Pull request closed without merging**